### DISCUSSION TITLE
Template issues

### DISCUSSION THREAD

**Discussion by jeromelaban** [unoplatform/uno#11812] [Q&A]
[_From Discord_:](https://discord.com/channels/372137812037730304/550416151172087808/1089753216423579739)

Anyone know why I only have Uno Platform App UWP template (snapshot of my avail templates, dark theme screenshot) and not Uno Platform App WinUI as shown in screen shot from https://platform.uno/docs/articles/getting-started-tutorial-2.html ?

I followed https://platform.uno/docs/articles/get-started-vs-2022.html

![image](https://user-images.githubusercontent.com/5839577/227951673-3eeff26a-091e-4e72-9a72-973a647c8d3a.png)

![image](https://user-images.githubusercontent.com/5839577/227951716-213f7c62-a4c9-41b4-907f-05dcae07eb90.png)

**Comment by jeromelaban** [comment]
Thanks for the report. Which of the screenshots is from your own installation?

**Reply by BinkersSoft** [reply]
As noted in my original message that you included in the opening post of this issue, it is the second screenshot with the dark theme.

I am also attaching the screenshot of the available templates when I filter by WinUI.
![Capture](https://user-images.githubusercontent.com/25735478/227987465-bd8b273d-9aef-4e56-a21e-67b5e6179698.png)

**Reply by jeromelaban** [reply]
Thanks. If you filter by Uno Platform, what do you see?

**Reply by BinkersSoft** [reply]
![Capture](https://user-images.githubusercontent.com/25735478/227989201-bf2682f8-a61f-4434-b110-caf09260fc42.JPG)

**Reply by jeromelaban** [reply]
I see. Thanks. I don't believe we're mentioning anywhere that the project filter should be set to WinUI, but that's something we can change to make it clearer. You can select "Uno Platform" in "All Project Types" as well.

Given this filter, do you have any additional issues when creating a new app?

**Reply by BinkersSoft** [reply]
The confusion is the image in the Getting Started tutorial shows that I should be choosing Uno Platform App that has WinUI in the platform list but I only see Uno Platform App with UWP in the platform list. It is shown in the light themed image in my original post. I also linked where on the Uno website I got it from.

**Reply by BinkersSoft** [reply]
This is on a fresh install of Windows 10 22H2 Build 19045.2788 and Visual Studio 2022 17.5.3 and I only added the Uno Platform extension to get the templates. I am also attaching the vsconfig exported from the Visual Studio Installer and the output of uno-check.

[vsconfig.txt](https://github.com/unoplatform/uno/files/11080470/vsconfig.txt)
[uno_check_output.txt](https://github.com/unoplatform/uno/files/11080474/uno_check_output.txt)

**Reply by BinkersSoft** [reply]
This is on a fresh install of Windows 10 22H2 Build 19045.2788 and Visual Studio 2022 17.5.3 and I only added the Uno Platform extension to get the templates.

**Reply by jeromelaban** [reply]
Thanks. It's indeed very curious that you're not getting all the templates by using the filter. Have you installed the dotnet new templates on the command line by any chance? (either the new uno.templates package or the older Uno.ProjectTemplates.Dotnet package)

**Reply by BinkersSoft** [reply]
When I first noticed not all the templates are visible I did try dotnet new install Uno.Templates to see if that would fix it and there was no change to the list of available templates.

**Reply by BinkersSoft** [reply]
In case this will help. Here is the output of dotnet --info

.NET SDK:
 Version:   7.0.202
 Commit:    6c74320bc3

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.19045
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\7.0.202\

Host:
  Version:      7.0.4
  Architecture: x64
  Commit:       0a396acafe

.NET SDKs installed:
  7.0.202 [C:\Program Files\dotnet\sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 6.0.15 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 7.0.4 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 6.0.15 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 7.0.4 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 6.0.15 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 7.0.4 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

Other architectures found:
  x86   [C:\Program Files (x86)\dotnet]
    registered at [HKLM\SOFTWARE\dotnet\Setup\InstalledVersions\x86\InstallLocation]

Environment variables:
  Not set

global.json file:
  Not found

Learn more:
  https://aka.ms/dotnet/info

Download .NET:
  https://aka.ms/dotnet/download

**Reply by BinkersSoft** [reply]
Using either cli command produce the same output:

The following template packages will be installed:
   Uno.Templates

Uno.Templates (version 4.8.8) is already installed, it will be replaced with latest version.
Uno.Templates::4.8.8 was successfully uninstalled.
Success: Uno.Templates::4.8.8 installed the following templates:
Template Name                        Short Name           Language  Tags
-----------------------------------  -------------------  --------  ------------------------------------------------------------------------------------------
Cross-Platform Library               unolib               [C#]      Cross-platform/Uno Platform/Library/Android/iOS/Windows/macOS/Linux/WebAssembly
Cross-Platform UI Tests Library      unoapp-uitest        [C#]      Cross-platform/Uno Platform/UITest/Android/iOS/macOS/WebAssembly
Cross-Runtime Library (WinUI)        unolib-crossruntime  [C#]      Multi-platform/Uno Platform/Android/iOS/macOS/Mac Catalyst/Windows/Linux/WebAssembly/WinUI
Multi-Platform App (.NET 6, UWP)     unoapp-uwp-net6      [C#]      Multi-platform/Uno Platform/Android/iOS/Windows/macOS/Linux/WebAssembly
Multi-Platform App (Uno.Extensions)  unoapp               [C#]      Multi-platform/Uno Platform/Android/iOS/macOS/Mac Catalyst/Windows/Linux/WebAssembly/WinUI
Multi-Platform App (Xamarin, UWP)    unoapp-uwp           [C#]      Multi-platform/Uno Platform/Android/iOS/Windows/macOS/Linux/Tizen/WebAssembly

**Reply by BinkersSoft** [reply]
That looks a big ugly. I put the output into a text file to make easier for you to read.
[template install via cli output.txt](https://github.com/unoplatform/uno/files/11080771/template.install.via.cli.output.txt)

**Reply by jeromelaban** [reply]
Thanks. So you installed the templates and you did not have them when you created the screenshots above, is that correct?

**Reply by BinkersSoft** [reply]
Correct

**Comment by BinkersSoft** [comment]
Gentle ping to see if a solution has been found or is in progress.

**Reply by jeromelaban** [reply]
A solution has been found, we'll be publishing an updated build. That being said, it's only a matter of filtering and it should not be preventing you from creating an app. Is this still the case?

**Reply by BinkersSoft** [reply]
We are not prevented in creating an app. Not having the full set of templates limits us from transitioning away from UWP to WinUI which we understand to be the direction new app development is going.

**Reply by jeromelaban** [reply]
Got it. Thanks for the update.
